manage

```diff
@@ -445,82 +447,91 @@ $app"
   #Used to track which line of $queue is currently being dealt with.
   current_line_num=1
   
+  sourced_updater=0 #updater script may need to be sourced if an update/refresh action is added to the queue. This prevents it from being sourced multiple times.
+  
   queue=''
   IFS=$'\n'
   while true;do #repeat until nothing is left in the queue
     
     #check for new actions to be executed
     echo -n > "${DIRECTORY}/data/manage-daemon/queue" & #ensure that the pipe is in write mode to prevent cat from hanging if the queue file is empty
     new_lines="$(tac "${DIRECTORY}/data/manage-daemon/queue")" # tac reverses the order of the list. a plain cat of the file will give the newest item in the queue first.
-
+    
+    if [ "$sourced_updater" == 0 ] && grep -q "update\|refresh" <<<"$new_lines" ;then
+      source "${DIRECTORY}/updater" source
+      sourced_updater=1
+    fi
+    
     #keep track of all actions for this session with the $queue variable
     if [ -z "$queue" ];then
       queue="$new_lines"
     elif [ ! -z "$new_lines" ];then # add new_lines to queue if new_lines is not empty
       queue+=$'\n'"$new_lines"
+      # reorder queue list to prioritize app refresh and file update actions
+      queue="$(reorder_list "$queue")"
     fi
-    # reorder queue list to prioritize app refresh and file update actions
-    queue="$(reorder_list "$queue")"
     #echo "length of queue is $(echo "$queue" | wc -l)"
     
-    if [ "${current_line_num}" -le "$(echo "$queue" | wc -l)" ];then #if we have not iterated through the entire queue length
-      #echo "current position in queue is ${current_line_num}"
-      #echo "queue is '$queue'"
-      
-      line="$(echo "$queue" | sed -n "${current_line_num}"p)"
-      #echo "Now handling request: '$line'"
-      
-      #indicate current action in current line of $queue
-      queue="$(echo "$queue" | sed "${current_line_num}s/$/;in-progress/")"
-      
-      #get first word of this line - the action. Subsequent words are the name of the app.
-      action="$(echo "$line" | awk -F';' '{print $1}')"
-      app="$(echo "$line" | awk -F';' '{print $2}')"
-      
-      #Set terminal title
-      echo -ne "\e]0;${action^}ing ${app}\a" | sed 's/Updateing/Updating/g'
-      
-      #refresh the list in queue-viewer window
-      write_list "$queue"
-      
-      #echo "Action is $action, app is $app"
-      
+    #exit loop if queue is complete and no new actions were added to the queue
+    if [ "${current_line_num}" -gt "$(echo "$queue" | wc -l)" ];then
+      break
+    fi
```

Comment:
I personally prefer `if` statements that execute if true rather than break out of the loop. however I understand why you made this change in this case (to get rid of the else statement) so I will let it slide